### PR TITLE
[Bifrost] WaitForTail message implementation

### PR DESCRIPTION
**Pull request by AhmedSoliman**
A new message that allows Bifrost to wait for a logserver to catch up to a certain local/global tail, or for the node to be sealed.

---
[//]: # (BEGIN SAPLING FOOTER)
Stack created with [Sapling](https://sapling-scm.com). Best reviewed with [ReviewStack](https://reviewstack.dev/restatedev/restate/pull/2012).
* #2019
* __->__ #2012
* #2008

### CODE REVIEW

File: crates/bifrost/src/providers/replicated_loglet/tasks/seal.rs [hunk 1]
```diff
@@ -140,6 +140,8 @@ impl<T: TransportConnect> SealSingleNode<T> {
             match self.do_seal().await {
                 Ok(res) if res.body().sealed || res.body().status == Status::Ok => {
                     let _ = tx.send((self.node_id, res.body().local_tail));
+                    self.known_global_tail
+                        .notify_offset_update(res.body().header.known_global_tail);
```
Comment (muhamadazmy) [inline crates/bifrost/src/providers/replicated_loglet/tasks/seal.rs:144]: Just to make sure I got this correctly. This is to make sure Clients that initiated the seal still learn about the last global tail know by the log server?

Comment (AhmedSoliman) [reply crates/bifrost/src/providers/replicated_loglet/tasks/seal.rs:144]: That's correct.

**Review by muhamadazmy — APPROVED** [review]

File: crates/log-server/src/network.rs [hunk 2]
```diff
@@ -217,6 +234,15 @@ impl RequestPump {
         trace!("All loglet workers have terminated");
     }
 
+    fn on_wait_for_tail(worker: &LogletWorkerHandle, msg: Incoming<WaitForTail>) {
+        if let Err(msg) = worker.enqueue_wait_for_tail(msg) {
+            // worker has crashed or shutdown in progress. Notify the sender and drop the message.
+            if let Err(e) = msg.to_rpc_response(TailUpdated::empty()).try_send() {
+                debug!(?e.source, peer = %e.original.peer(), "Failed to respond to WaitForTail message with status Disabled due to peer channel capacity being full");
```
Comment (tillrohrmann) [inline crates/log-server/src/network.rs:241]: The debug message seems not aligned with the above comment and the response we are sending.

**Review by tillrohrmann — APPROVED** [review]
Nice work @AhmedSoliman :-) The changes look good to me. +1 for merging.

File: crates/log-server/src/network.rs (hunk 2 above)
Comment (AhmedSoliman) [reply crates/log-server/src/network.rs:241]: The debug message is printed when we fail to send the response. The above comment is about why we are actually trying to respond now.

Comment (tillrohrmann) [reply crates/log-server/src/network.rs:241]: True, makes sense 👍

**Pull request merged**